src/Controller/JobController.php

```diff
@@ -0,0 +1,327 @@
+     * read a job by the substring of the name
+     *
+     * @param  mixed $name
+     * @return Response
+     */
+    public function readJobsByNameLike(string $name): Response
+    {
+        try {
+            $this->jobValidator->nameIsValid($name);
+            $rows = $this->jobsRepository->selectByNameLike($name);
+            $jobsArr = [];
+            foreach ($rows as $job) {
+                $jobsArr[] = [
+                    'id' => $job->getId(),
+                    'name' => $job->getName(),
+                    'description' => $job->getDescription(),
+                    'createdAt' => $job->getCreatedAt(),
+                    'company' => [
+                        'id' => $job->getCompany()->getId(),
+                        'company' => $job->getCompany()->getName()
+                        
+                    ]
+                ];
+            }
+            
+            return new JsonResponse([
+                'results' => [
+                    'error' =>false,
+                    'jobs' => $jobsArr
+                ]
+            ]);
+        } catch (\Exception $e) {
+            return new JsonResponse(
+                [
+                    'results' => [
+                        'error' => true,
+                        'message' => $e->getMessage()
+                    ]
+                ]
+            );
+        }
+    }
+
+    /**
+     * updates a job from the database
+     *
+     * @param  mixed $id
+     * @param  mixed $request
+     * @return Response
+     */
+    public function updateJob(int $id, Request $request): Response
+    {
+        try {
+            $this->jobValidator->idIsValid($id);
+            $name = $request->get('name');
+            $this->jobValidator->nameIsValid($name);
+            $requestParams = $request->query->all();
+            $updateResult = $this->jobsRepository->update($id, $requestParams);
+            return new JsonResponse(
+                [
+                    'results' => [
+                        'error' => false,
+                        'rows_updated' => $updateResult
+                    ]
+                ]
+            );
+        } catch (\Exception $e) {
+            return new JsonResponse(
+                [
+                    'results' => [
+                        'error' => true,
+                        'message' => $e->getMessage()
+                    ]
+                ]
+            );
+        }
+    }
+
+    /**
+     * delete a job from the database
+     *
+     * @param  mixed $id
+     * @return Response
+     */
+    public function deleteJob(int $id): Response
+    {
+        try {
+            $this->jobValidator->idIsValid($id);
+            $deletedId = null;
+            $jobToDelete = $this->jobsRepository->find($id);
+            if (is_null($jobToDelete)) {
+                throw new \InvalidArgumentException('Could not find job with id: ' . $id);
+            }
+            
+            if (!empty($jobToDelete)) {
```

Comment:
You do not need to check if $jobToDelete is empty because it will never be empty at line 305 because this check is done at line 301. 
Do not need a new variable  $deletedId because it just keeps what is in $jobToDelete variable.

You can replace this issues with:

![image](https://user-images.githubusercontent.com/19302124/207373763-a6ffc9a8-c55e-4afd-9282-60d9cbd69354.png)